src/validation/coinbase.cairo

```diff
@@ -66,9 +67,47 @@ pub fn validate_coinbase(
         if witness != @"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0" {
             return Result::Err("Wrong coinbase witness");
         }
-    }
 
-    // TODO: validate BIP-141 segwit output
+        // validate BIP-141 segwit output
+        if *tx.is_segwit {
+            let outputs = *tx.outputs;
+            let mut is_wtxid_commitment_present: bool = false;
+            let mut extracted_wtxid_commitment: ByteArray = "";
+            let mut i = 0;
+            let mut x = 6;
+
+            while i < outputs.len() {
```

Comment:
If multiple outputs contain witness commitment then the one with the maximum vout is taken.  
You can use `while let Option::Some(output) = outputs.pop_back()` loop for reverse iteration.